ast/desugar/Desugar.cc

```diff
@@ -228,13 +237,29 @@ ExpressionPtr mergeStrings(DesugarContext dctx, core::LocOffsets loc,
     }
 }
 
+// In DString, the `#{...}` are all wrapped in parser::Begin nodes.
+// Normally, we have handling for parser::Begin nodes that treats the loc of such a node to include
+// the enclosing brackets. So e.g. the loc of `(x)` would be length 3.
+// But the parser also uses Begin nodes for `#{...}`, and for those we don't want the loc to include
+// the braces (think: we don't want a `T.must` to look like `"T.must(#{x})"`, we want it to look
+// like `"#{T.must(x)}"`, etc.)
+// So we have this skipSingletonBegin to skip over these nodes before desugaring them in DString
```

Comment:
Judging from the commit history, we do have tests around this?  I didn't see any for `T.must` and the like, but it does look like we have some autocorrects applying inside interpolated strings?

Reply:
I think the trick is that the example I gave in the comment was hypothetical—if string interpolation required the argument to be non-nil, we could write a test for this

But as it is, I don't think we can easily write a test for when the autocorrect would apply to the whole range of the interpolated string.